.github/workflows/ci.yaml

```diff
@@ -66,73 +66,38 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
   build:
-    name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v8
+    name: Build charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v11.0.1
+    with:
+      cache: true
 
   integration-test:
     strategy:
       fail-fast: false
       matrix:
-        tox-environments:
-          - charm-integration
-          - ha-integration
-          - relation-integration
-          - legacy-integration
-          - tls-integration
-          - backup-integration
-          - metric-integration
-          - sharding-integration
-          - sharding-relation-integration
-          - sharding-race-conditions-integration
-          - sharding-mongos-integration
-    name: ${{ matrix.tox-environments }}
+        juju:
+          - agent: 3.1.6
+            libjuju: ^3
+            allure: true
```

Comment:
JFYI, this is still WIP and API here is not stable. You ALLURE on your own risk for now. :-D